### PR TITLE
samples: mesh: single timer to handle all transitions, upgrade the implementation of Gen. Move Message handlers & bug fixes

### PR DESCRIPTION
**Pull request by vikrant8052**
Commit 1:
This commit upgrade the implementation to reduce complexity in
algorithm. Previously there was different timers for lightness
as well as temperature transitions. But at time only single
transition is remain activated & hence there is no need
of multiple timer for each entity.

commit 2:
If Node Lightness is in downward transition & received Generic
onoff message to set state equal to 1 then Node transaction get
stopped in between. Ideally it should reach to default lightness
value if it is non-zero. This commit has solved this bug.

commit 3:
Remove redundant code from implementation.

commit 4:
After receiving Generic Level Move set/set_unack message, generic
level state should move towards positive or negative extreme end
(which is depend upon sign of delta value) till it not get interruped.
This commit has introduced this feature.

commit 5:
Replaced magic numbers with constant defined in stdint.h

commit 6:
Synchronizing the init procedure to initialize the mesh state only after
the settings have been loaded.

Signed-off-by: Vikrant More vikrant8051@gmail.com

### CODE REVIEW

**Comment by vikrant8051** [comment]
@jhedberg 
I have close https://github.com/zephyrproject-rtos/zephyr/pull/15233 because some
rebase issue which was beyond my scope. I tried to solve it & it went full out of control.

Kindly consider this new PR & help me to merge this.

File: samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/main.c [hunk 1]
```diff
@@ -198,17 +197,17 @@ void main(void)
 	ps_settings_init();
 
 	/* Initialize the Bluetooth Subsystem */
-	err = bt_enable(bt_ready);
+	err = bt_enable(NULL);
 	if (err) {
 		printk("Bluetooth init failed (err %d)\n", err);
```
Comment (jhedberg) [inline samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/main.c]: @vikrant8051 you had earlier a return statement here. Why did you remove it? I suppose none of the subsequent code in main() makes sense if bt_enable() fails. Also, if/when you add back the return statement, the else branch can be removed (i.e. call bt_ready on the same level as e.g. light_default_status_init)

Comment (vikrant8051) [reply samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/main.c]: Done !!

**Review by jhedberg — APPROVED** [review]

**Comment by vikrant8051** [comment]
@jhedberg Thank You !! :)

**Pull request merged**